packages/flutter/lib/src/services/platform_views.dart

```diff
@@ -61,6 +61,25 @@ class PlatformViewsRegistry {
 /// The `id` parameter is the platform view's unique identifier.
 typedef PlatformViewCreatedCallback = void Function(int id);
 
+/// Indicates the reason for a request to move keyboard navigation out of a platform view back into the widget tree.
+enum NavigationReason {
+  /// Non-directional
+  kProgrammatic,
```

Comment:
I think there's got to be a clearer name but can't think of anything super obvious. `kNonDirectional` as in your comment would be better but what we're after, is an immediate switch to some other element. @gspencergoog authored a lot of the focus code in the framework and might have some ideas. @goderbauer is good at naming things too!

Reply:
`PROGRAMMATIC` is the name used by WebView. We need not use the same naming scheme, though that was the rationale.

Reply:
Makes sense. I'd avoid tying the naming to a specific plugin's underlying API. Seems like they maybe had the same debate over what the heck to call this :)

Reply:
Do you have any input on whether you think we should go with this PR's approach or https://github.com/flutter/flutter/pull/144569?